quickwit/quickwit-actors/src/spawn_builder.rs

```diff
@@ -306,6 +350,8 @@ async fn actor_loop<A: Actor>(actor: A, inbox: Inbox<A>, ctx: ActorContext<A>) -
         actor_env.process_messages().await
     };
 
+    // TODO the no advance time guard for finalize has a race condition. Ideally we would
```

Comment:
Good point indeed.
We could have a guard in the returned error by `process_messages`.
One note here, I would have probably kept a guard in `process_all_available_messages` and avoid keeping a guard in each envelope. I may have missed subtleties here though.

Reply:
That would have been incorrect... We want to avoid jump in time as soon as there are messages scheduled for processing.